src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java

```diff
@@ -442,10 +444,16 @@ public boolean stopRun(String controllerApiHttpAddress, UUID runId, String sched
                     listener.error(String.format(Messages.STOP_RUN_FAIL,scheduleTitle,runId.toString()));
 
             }
-        } catch (Exception e)
+        } 
+        
+        catch (RuntimeException e) {
+			throw new RuntimeException(e);
+		}
+        catch (Exception e)
         {
             listener.error(String.format(Messages.STOP_RUN_FAIL,scheduleTitle,runId.toString()));
             listener.error(e.getMessage());
+           throw e;
```

Comment:
throwing this error leads to losing some logic in the calling method, check `CollectScheduleRunResults`